base/gmp.jl

```diff
@@ -41,23 +41,30 @@ type BigInt <: Integer
     alloc::Cint
     size::Cint
     d::Ptr{Limb}
-    function BigInt()
-        b = new(zero(Cint), zero(Cint), C_NULL)
-        ccall((:__gmpz_init,:libgmp), Void, (Ptr{BigInt},), &b)
-        finalizer(b, _gmp_clear_func)
-        return b
+    function BigInt(::Val{:allocbits}, nbits)
```

Comment:
If it's just going to be internal, we can use a name like `BigInt_withsize` and switch to a keyword arg later when they are fast.